client/src/pages/index.tsx

```diff
@@ -1,38 +1,54 @@
-import SocketProvider, { SocketContext } from '@/components/websocket/websocket';
-import { useContext } from 'react';
+import  { getSocketContext } from '@/components/websocket/websocket';
+import { useTurnContext } from '@/components/turnProvider/turnProvider';
 import { useRouter } from 'next/router';
 
-function welcomeChild() {
-    let socket = useContext(SocketContext);
+export default function welcomeChild() {
+    let socket = getSocketContext();
+    const {setTurn} = useTurnContext()
     const router = useRouter();
-
+    if (socket === undefined) {
+        return <div>Undefined socket</div>
+    }
     socket.on('sessionCreated', function(data) {
         sessionStorage.setItem("sessionId", data.sessionId);
     })
 
     function handleCellClick() {
         let data = {row: 0, col: 0};
+        if (socket === undefined) {
+            console.log("no socket")
+            return
+        }
```

Comment:
Why are we checking this in each function?

Reply:
each function could use the socket object which can be undefined, if not used in a provider